source/common/http/conn_manager_utility.cc

```diff
@@ -78,7 +78,16 @@ ServerConnectionPtr ConnectionManagerUtility::autoCreateCodec(
 ConnectionManagerUtility::MutateRequestHeadersResult ConnectionManagerUtility::mutateRequestHeaders(
     RequestHeaderMap& request_headers, Network::Connection& connection,
     ConnectionManagerConfig& config, const Router::Config& route_config,
-    const LocalInfo::LocalInfo& local_info) {
+    const LocalInfo::LocalInfo& local_info, const StreamInfo::StreamInfo& stream_info) {
+
+  if (!config.earlyHeaderMutationExtensions().empty()) {
```

Comment:
nit: this empty check isn't needed. The for loop will just not execute below if empty.